tests/protocol/test_hostplugin.py

```diff
@@ -129,65 +129,169 @@ def _validate_hostplugin_args(self, args, goal_state, exp_method, exp_url, exp_d
+    @patch("azurelinuxagent.common.protocol.wire.StatusBlob.upload", return_value=False)
+    @patch("azurelinuxagent.common.protocol.hostplugin.HostPluginProtocol._put_page_blob_status")
+    @patch("azurelinuxagent.common.protocol.wire.WireClient.update_goal_state")
+    def test_default_channel(self, patch_update, patch_put, patch_upload, _):
         """
         Status now defaults to HostPlugin. Validate that any errors on the public
         channel are ignored.  Validate that the default channel is never changed
         as part of status upload.
         """
         test_goal_state = wire.GoalState(WireProtocolData(DATA_FILE).goal_state)
         status = restapi.VMStatus(status="Ready", message="Guest Agent is running")
-        with patch.object(wire.HostPluginProtocol,
-                          "ensure_initialized",
-                          return_value=True):
-            with patch.object(wire.StatusBlob, "upload", return_value=False) as patch_upload:
-                with patch.object(wire.HostPluginProtocol,
-                                  "_put_page_blob_status") as patch_put:
-                    wire_protocol_client = wire.WireProtocol(wireserver_url).client
-                    wire_protocol_client.get_goal_state = Mock(return_value=test_goal_state)
-                    wire_protocol_client.ext_conf = wire.ExtensionsConfig(None)
-                    wire_protocol_client.ext_conf.status_upload_blob = sas_url
-                    wire_protocol_client.ext_conf.status_upload_blob_type = page_blob_type
-                    wire_protocol_client.status_blob.set_vm_status(status)
-                    wire_protocol_client.upload_status_blob()
-                    self.assertEqual(patch_upload.call_count, 0)
-                    self.assertTrue(patch_put.call_count == 1,
-                                    "Fallback was not engaged")
-                    self.assertTrue(patch_put.call_args[0][0] == sas_url)
-                    self.assertFalse(wire.HostPluginProtocol.is_default_channel())
 
-    def test_fallback_failure(self):
+        wire_protocol_client = wire.WireProtocol(wireserver_url).client
+        wire_protocol_client.get_goal_state = Mock(return_value=test_goal_state)
+        wire_protocol_client.ext_conf = wire.ExtensionsConfig(None)
+        wire_protocol_client.ext_conf.status_upload_blob = sas_url
+        wire_protocol_client.ext_conf.status_upload_blob_type = page_blob_type
+        wire_protocol_client.status_blob.set_vm_status(status)
+
+        # act
+        wire_protocol_client.upload_status_blob()
+
+        # assert direct route is not called
+        self.assertEqual(0, patch_upload.call_count, "Direct channel was used")
+
+        # assert host plugin route is called
+        self.assertEqual(1, patch_put.call_count, "Host plugin was not used")
+
+        # assert update goal state is only called once, non-forced
+        self.assertEqual(1, patch_update.call_count, "Unexpected call count")
```

Comment:
Assert that forced was set to False.

Reply:
That is what the check in L163 does - validate that no parameters were passed to the method, which defaults to False.